src/Scope/Core.agda

```diff
@@ -50,6 +50,17 @@ instance
   Monoid.mappend iMonoidScope = scopeMappend
   Monoid.mconcat iMonoidScope = scopeMConcat
 
+data RScope (@0 name : Set) : Set where
+  Nil : RScope name
+  Cons : (@0 x : name) (s : RScope name) → RScope name
```

Comment:
Here you use `@0` but the definition of regular scopes uses `Erased`, perhaps it's better to keep this uniform.